test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js

```diff
@@ -0,0 +1,188 @@
+const startBrowserTimeoutMS = config.get( 'startBrowserTimeoutMS' );
+const screenSize = driverManager.currentScreenSize();
+const host = dataHelper.getJetpackHost();
+const gutenbergUser =
+	process.env.COBLOCKS_EDGE === 'true' ? 'searchSimpleSiteEdgeUser' : 'gutenbergSimpleSiteUser';
+
+describe( `[${ host }] Calypso Gutenberg Editor: Search Block (${ screenSize })`, function () {
+	this.timeout( mochaTimeOut );
+	let driver;
+
+	before( 'Start browser', async function () {
+		this.timeout( startBrowserTimeoutMS );
+		driver = await driverManager.startBrowser();
+	} );
+
+	describe( 'Insert Search Block and confirm it exists in te edit page: @parallel', function () {
+		it( 'Can log in', async function () {
+			this.loginFlow = new LoginFlow( driver, gutenbergUser );
+			return await this.loginFlow.loginAndStartNewPost( null, true );
+		} );
+
+		it( 'Can Insert the Search block', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			await gEditorComponent.addBlock( 'Search' );
+			return await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search' )
+			);
+		} );
+
+		it( 'Can see search block elements', async function () {
+			const textLocator = By.css( '.wp-block-search__label' );
+			await driverHelper.waitUntilElementLocatedAndVisible( driver, textLocator );
+			await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__input' )
+			);
+
+			return driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__button' )
+			);
+		} );
+
+		it( 'Can publish and view content', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			// We need to save the post to get a stable post slug for the block's `url` attribute.
+			// See https://github.com/godaddy-wordpress/coblocks/issues/1663.
+			await gEditorComponent.ensureSaved();
+			return await gEditorComponent.publish( { visit: true } );
+		} );
+
+		it( 'Can see the Search block in our published post', async function () {
+			return await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search' )
+			);
+		} );
+	} );
+
+	describe( 'Verify Search block Menubar and Settings Side bar: @parallel', function () {
+		it( 'Can log in', async function () {
+			this.loginFlow = new LoginFlow( driver, gutenbergUser );
+			return await this.loginFlow.loginAndStartNewPost( null, true );
+		} );
+
+		it( 'Can Insert the Search block', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			await gEditorComponent.addBlock( 'Search' );
+			return await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search' )
+			);
+		} );
+
+		it( 'Can Toggle Search Block', async function () {
+			const textLocator = By.css( '.wp-block-search__input' );
```

Comment:
A lot of the settings in this section might work better as a "Page Object" that contains the settings for the search block. It would really clean this up and let you re-use things.

Reply:
I will add these to Page objects. Just confirming, this is where all the page objects are right? /wp-calypso/test/e2e/lib/pages

Reply:
Yeah, they are all in lib somewhere. Blocks might be better in gutenberg/blocks

Reply:
@bsessions85  - Is there any README page that gives more information about how the Page Objects set up in calypso . I was trying to read this https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/docs/style-guide.md#page-objects and follow up but what's mentioned in "Don't Do" , some flows/components do that.

Reply:
@sowmya2709 That is really all the documentation that we have. The key in that readme section is to make sure you're using the 'expect' or 'visit' method from the test and not just creating a new instance of the object. I recommend finding one in there and using it as a template.

Reply:
The documents are present but there is no guarantee that they are up to date. The code itself, and how it's structured should be the primary form of documentation.

Take a look at some of the page objects that exist in the repo for inspiration. A good one is [posts-page](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/pages/posts-page.js) and another is [login-page](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/pages/login-page.js).

Reply:
Hi @worldomonation, Thanks for these references